### PR TITLE
back button fix #322

### PR DESCRIPTION
**Pull request by jasontsh**
Fixed the back button problem inside the registrar!

### CODE REVIEW

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/RegistrarTab.java [hunk 1]
```diff
@@ -52,6 +54,20 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+            public boolean onKey(View v, int keyCode, KeyEvent event) {
+                if (keyCode == KeyEvent.KEYCODE_BACK && event.getAction() == KeyEvent.ACTION_DOWN && fragment != null) {
+                    FragmentManager fragmentManager = mActivity.getSupportFragmentManager();
+                    fragmentManager.beginTransaction().remove(fragment).commit();
+                    fragment = null;
+                    return true;
+                }
+                return false;
+            }
+        });
```
Comment (adelq) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/RegistrarTab.java]: This should definitely not be in the `MainActivity onCreateView`, far too much complexity that should be refactored.

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/RegistrarTab.java [hunk 2]
```diff
@@ -72,11 +88,10 @@ public void onListItemClick(ListView l, View v, int position, long id) {
         args.putBoolean(getString(R.string.registrar_search), favorites);
         fragment.setArguments(args);
 
-        FragmentManager fragmentManager = getActivity().getSupportFragmentManager();
+        FragmentManager fragmentManager = mActivity.getSupportFragmentManager();
         fragmentManager.beginTransaction()
-                .replace(frameID, fragment)
+                .add(frameID, fragment)
```
Comment (adelq) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/RegistrarTab.java]: Are you sure this doesn't create new problems with redraw and so on? I think this needs more testing

Comment (jasontsh) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/RegistrarTab.java]: When the back button is pressed, the fragment manager removes the fragment, and so if we use replace, the old one would be gone as well.

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/RegistrarTab.java (hunk 1 above)
Comment (dlakata) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/RegistrarTab.java]: Yeah, I agree that this is rather detailed for the CourseFragment

**Comment by dlakata** [comment]
Back button still has some weird behavior: search for a course, select a result, use the Android back button to go back, and the action bar back button remains and is unusable

**Pull request merged**